app/assets/javascripts/calendar/calendar.js

```diff
@@ -0,0 +1,63 @@
+$(function () {
+    events.forEach(function (block) {
+        var eventWidth = getEventWidth(block);
+
+        block.forEach(function (event) {
+            var tableCell = getEventTableCell(event);
+            var eventContainer = tableCell.children('.event-container');
+            var eventsInThisContainer = block.filter(function (e) {
+                return e.start_time == event.start_time;
+            }).length;
+
+            var width = eventsInThisContainer * eventWidth,
+                offset = (block.indexOf(event) - eventsInThisContainer + 1) * eventWidth;
+            eventContainer.css({width: width, 'margin-left': offset});
+
+            var length = (new Date(event.end_time).getTime() - new Date(event.start_time).getTime()) / 1000 / eventTimeGranularity;
```

Comment:
How about extracting this:

`new Date(event.end_time).getTime() - new Date(event.start_time).getTime()) / 1000`

to a method called `eventLengthInSeconds`?